### DISCUSSION TITLE
Cannot generate link

### DISCUSSION THREAD

**Discussion by cvg095** [LSPosed/MagiskOnWSALocal#464] [General]
When I finish selecting options this error always occurs. I am using Ubuntu for Windows. Please help. Thank you

COMMAND_LINE=--arch x64 --release-type retail --magisk-ver stable --gapps-brand MindTheGapps --gapps-variant pico --remove-amazon --root-sol magisk --compress    --compress-format 7z
INFO: Architecture: x64
INFO: Release Type: retail
INFO: Magisk Version: stable
INFO: GApps Brand: MindTheGapps
INFO: GApps Variant: pico
INFO: Root Solution: magisk
INFO: Compress Format: 7z
Build: RELEASE_TYPE=Retail
Generate Download Links
Traceback (most recent call last):
  File "/home/csman/MagiskOnWSALocal/scripts/generateWSALinks.py", line 32, in <module>
    from requests import Session
ModuleNotFoundError: No module named 'requests'
Build: an error has occurred, exit

Cleanup Work Directory

![image](https://user-images.githubusercontent.com/114903208/228134268-798b9f1c-6c47-4240-bf88-45091a90324d.png)

**Comment by ChinmayaGit** [comment]
hi I am not for sure but select no on (compress output).

**Comment by cvg095** [comment]
Hi found a solution, thank you

need to run this first

sudo apt-get update
 
sudo apt-get upgrade
